drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -1315,13 +1325,7 @@ vc4_hdmi_encoder_mode_valid(struct drm_encoder *encoder,
 	     (mode->hsync_end % 2) || (mode->htotal % 2)))
 		return MODE_H_ILLEGAL;
 
-	if ((mode->clock * 1000) > vc4_hdmi->variant->max_pixel_clock)
-		return MODE_CLOCK_HIGH;
-
-	if (vc4_hdmi->disable_4kp60 && vc4_hdmi_mode_needs_scrambling(mode))
-		return MODE_CLOCK_HIGH;
-
-	return MODE_OK;
+	return vc4_hdmi_encoder_clock_valid(vc4_hdmi, mode->crtc_clock * 1000);
```

Comment:
This should be `mode->clock` not `mode->crtc_clock`.
It's the cause of blank screen when booting LE at 4kp30.
Logging shows `mode->clock` is set to plausible value but `mode->crtc_clock` is often `0` here.

Reply:
Indeed, it's filled by drm_mode_set_crtcinfo() which is called right after .mode_valid. I've fixed it.